### PR TITLE
Add dynamic sync tables guide

### PR DESCRIPTION
**Pull request by ekoleda-codaio**
In addition to the guide content, this PR also:

- Makes some tweaks to the Typeform sample.
- Updates the description for the dynamic sync tables samples.
- Standardizes on the term "function" to describe the functions used in various definitions.

Staged: https://coda-packs-sdk--1743.com.readthedocs.build/en/1743/guides/blocks/sync-tables/dynamic/

### CODE REVIEW

File: documentation/examples/dynamic-sync-table/README.md [hunk 1]
```diff
@@ -1,11 +1,3 @@
-Most sync tables have schemas that can be statically defined. For example, if you're writing a sync of a user's Google Calendar events, the structure of an Event from the Google Calendar API is well-known and you can write a schema for what your table should contain.
+Dynamic sync tables allow you to bring data from an external data source into your doc, even when the structure of that data is not known in advance. Instead of including a static schema in your sync table definition, you include a `getSchema` method that returns a schema based on the dataset the user selects. It can use the fetcher to make authenticated HTTP requests to an API so that you can determine the shape of the data.
```
Comment (jonathan-codaio) [inline documentation/examples/dynamic-sync-table/README.md:1]: This still uses "method", was that intentional?

File: documentation/examples/dynamic-sync-table/README.md [hunk 2]
```diff
@@ -1,11 +1,3 @@
...
-
-Alternatively, a sync table can be dynamic because the data source is specific to each instance of the table. If you were building a sync table to sync data from a Google Sheet, the data source would be the API url of a specific sheet. In this case, the sync table will be bound to a `dynamicUrl` that defines the data source. This url will be available to all of the formulas to implement the sync table in the sync context, as `context.sync.dynamicUrl`. To create a sync table that uses dynamic urls, you must implement the `listDynamicUrls` metadata formula in your dynamic sync table definition.
+Dynamic sync tables typically provide list all of the datasets that the user has access to, using the `listDynamicUrls` method. Once a user selects a dataset from the side panel it will be available to other method in `context.sync.dynamicUrl`. The dynamic sync table must also provide a user-visible URL for the selected dataset (using `getDisplayUrl`) and a name for the resulting table on the page (using `getName`).
```
Comment (jonathan-codaio) [inline documentation/examples/dynamic-sync-table/README.md:3]: Typos, "provide list" --> probably "provide a list" and "method" --> "methods"

File: docs/guides/blocks/sync-tables/dynamic.md [hunk 3]
```diff
@@ -1,7 +1,428 @@
 ---
-title: Dynamic schemas
+title: Dynamic sync tables
 ---
 
 # Creating sync tables with dynamic schemas
 
-TODO
+Sync tables are designed to bring records from an external data source into Coda, but sometimes those records don't have a predefined structure. For example, a task tracking application may allow each project to define it's own set of custom fields. To bring this type of data into Coda you can create a dynamic sync table, which allows the columns in the table to adapt to the dataset that is being synced.
```
Comment (jonathan-codaio) [inline docs/guides/blocks/sync-tables/dynamic.md:7]: "its" :)

File: docs/guides/blocks/sync-tables/dynamic.md [hunk 4]
```diff
@@ -1,7 +1,428 @@
...
+  formula: {
+    // Sync the rows from the dataset.
+    // ...
+  },
+});
+```
+
+More information the the various components of the dynamic sync table are described in the sections below.
+
+!!! bug
```
Comment (jonathan-codaio) [inline docs/guides/blocks/sync-tables/dynamic.md:57]: Ah, this is still a bug? Does it work to override the `connectionRequirement` if you don't want to use authentication?

File: docs/guides/blocks/sync-tables/dynamic.md [hunk 5]
```diff
@@ -1,7 +1,428 @@
...
+        value: project.links.self,
+      });
+    }
+    return results;
+  },
+  // ...
+});
+```
+
+The URL you choose to represent the dataset is arbitrary, and is not directly exposed to the user. Most Packs choose to use an API endpoint URL for ease of development. The URL shown to users is generated using the [`getDisplayUrl`](#display-url) method.
```
Comment (jonathan-codaio) [inline docs/guides/blocks/sync-tables/dynamic.md:92]: I think there might be better framing than saying the URL is arbitrary. It's true that you can put arbitrary strings for the dynamic url and that might be worth noting. But the spirit of the field is that you use it to specify an API entity to sync from, and the value very directly instructs your sync implementation on where to get data. So in that sense it's not arbitrary, in the common case the value needs to be something quite specific, though its form can vary as desired.

**Review by jonathan-codaio — APPROVED** [review]

File: documentation/examples/dynamic-sync-table/README.md (hunk 1 above)
Comment (ekoleda-codaio) [reply documentation/examples/dynamic-sync-table/README.md]: No, overlooked. Fixed.

File: documentation/examples/dynamic-sync-table/README.md (hunk 2 above)
Comment (ekoleda-codaio) [reply documentation/examples/dynamic-sync-table/README.md]: Done

File: docs/guides/blocks/sync-tables/dynamic.md (hunk 3 above)
Comment (ekoleda-codaio) [reply docs/guides/blocks/sync-tables/dynamic.md]: KHAAAAAAN! Done.

File: docs/guides/blocks/sync-tables/dynamic.md (hunk 4 above)
Comment (ekoleda-codaio) [reply docs/guides/blocks/sync-tables/dynamic.md:57]: Ya. Even with `connectionRequirement` of `None` you still get a **Connect** button that crashes the doc when you click it. I list the related bugs in the comment below.

File: docs/guides/blocks/sync-tables/dynamic.md (hunk 5 above)
Comment (ekoleda-codaio) [reply docs/guides/blocks/sync-tables/dynamic.md]: Fair enough, I've removed the use of the word "arbitrary" and rewritten it a little.

**Pull request merged**